File: src/testpub.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-


import rospy
from std_msgs.msg import Int32

def talker():
    pub = rospy.Publisher('direction', Int32, queue_size=10)
    rospy.init_node('testpub', anonymous=True)
    rate = rospy.Rate(1) # 초당 n번 발행
    while not rospy.is_shutdown():
        inputdat = input('input any direction: ')
        inputdata = int(inputdat)
        rospy.loginfo("input data : %d" %(inputdata))
        pub.publish(inputdata)
        rate.sleep()

if __name__ == '__main__':
    try:
        talker()
    except rospy.ROSInterruptException:
        pass

File: src/testpub.cpp
#include "ros/ros.h"
#include "std_msgs/Int32.h"
#include <string>
#include <iostream>
#include <cstdio>
#include <cstring>
#include <memory>
#include <unistd.h>
#include <sound_play/sound_play.h>

int main(int argc, char **argv){
  ros::init(argc, argv, "testpub");
  ros::NodeHandle n;
  ros::Publisher pub = n.advertise<std_msgs::Int32>("/direction", 10);
  ros::Rate loop_rate(10);

  int waydata;
  ROS_INFO("input data :");
  scanf("%d", &waydata);

  while(ros::ok()){
    std_msgs::Int32 msg;
    ROS_INFO("%d is input", msg.data);
    pub.publish(msg);
    ros::spin();
    loop_rate.sleep();
  }
  return 0;
}
